pkg/util/ipvs/ipvs_linux.go

```diff
@@ -85,7 +93,10 @@ func (runner *runner) GetVirtualServer(vs *VirtualServer) (*VirtualServer, error
 	if err != nil {
 		return nil, err
 	}
+	runner.mu.Lock()
 	ipvsSvc, err := runner.ipvsHandle.GetService(svc)
+	runner.mu.Unlock()
```

Comment:
Why not use 'defer' ?

Reply:
We could get the lock at the beginning of each function and use defer everywhere to make everything consistent. I chose to release the lock as quickly as possible. We do not need to keep the lock to call `toVirtualServer` in `GetVirtualServer` for instance.

I'm completely open to changing this if it makes the code easier to read and the optimization is not worth it.

Reply:
True.